lib/commands.coffee

```diff
@@ -51,12 +52,26 @@ class Commands
       atom.notifications.addError error.message, {detail: error.stack, dismissable: true}
 
   nextError: ->
-    if not @messages or (next = @messages.next()).done
-      next = (@messages = @linter.views.getMessages().values()).next()
-    return if next.done # There's no errors
-    message = next.value
-    return unless message.filePath
-    return unless message.range
+    if @index?
```

Comment:
Seems like we should really be indexing errors by some sort of hash of position and error message.  Otherwise changes to the full set of messages would not have a stable order.  This is of course assuming our linter messages have a stable order.

On that note we should really have a stable order, because it doesn't look like multiple linters actually have a stable order in our error processing.